custom_components/xiaomi_cloud_map_extractor/common/image_handler.py

```diff
@@ -243,31 +245,36 @@ def draw_func(draw: ImageDraw):
             coords = [point.x - r, point.y - r, point.x + r, point.y + r]
             draw.pieslice(coords, angle+90, angle-90, outline="black", fill=fill)
 
-        ImageHandler.__draw_on_new_layer__(image, draw_func)
+        use_transparency = (len(outline) > 3 or len(fill) > 3)
+        ImageHandler.__draw_on_new_layer__(image, draw_func, 1, use_transparency)
 
     @staticmethod
     def __draw_areas__(image: ImageData, areas, fill, outline):
         if len(areas) == 0:
             return
-        for area in areas:
-            def draw_func(draw: ImageDraw):
+        def draw_func(draw: ImageDraw):
```

Comment:
This will cause problems with overlapping areas

Reply:
Why do you say that? Can you explain?

Reply:
Only the latest one will be drawn on the intersection

Reply:
But that would happen I'm any case, right? Am I missing something?

Reply:
Before your change (`draw_func` inside `for`):
```
 ----
|    |
|   -|--
|  | |  |
 ----   |
   |    |
    ----
```
After your change (`for` inside `draw_func`):
```
 ----
|    |
|    |--
|    |  |
 ----   |
   |    |
    ----
```

Reply:
Ah yes... right. Back to how it was then...

Reply:
Ah yes, right... back to how it was, then...